src/api.ts

```diff
@@ -59,6 +59,126 @@ function addSkill(devId: number, newSkill: string): boolean {
   return false;
 }
 
+function updateSkill(
+  devId: number,
+  oldSkill: string,
+  newSkill: string,
+): boolean {
+  const developer = developers.find(
+    ({ id: existingDevId }) => existingDevId === devId,
+  );
+
+  if (!developer) {
+    return false;
+  }
+
+  const { skills } = developer;
+
+  if (
+    !skills.some((existingSkill) => existingSkill === newSkill) &&
+    skills.some((existingSkill) => existingSkill === oldSkill)
+  ) {
+    developer.skills[
+      developer.skills.findIndex((existingSkill) => existingSkill === oldSkill)
+    ] = newSkill;
+    return true;
+  }
+
+  return false;
+}
+
+function updateDeveloper(
+  devId: number,
+  updates: Partial<Omit<Developer, "id">>,
+) {
+  const developerIndex = developers.findIndex(
+    ({ id: existingDevId }) => existingDevId === devId,
+  );
+
+  if (developerIndex === -1) {
+    throw Error("Error: developer with devId does not exist");
+  }
+
+  developers[developerIndex] = {
+    ...developers[developerIndex],
+    ...updates,
+    id: developers[developerIndex].id,
+  };
+}
+
+function removeDeveloperByCondition(
+  callbackCondition: (developer: Developer) => boolean,
+) {
+  const deleted: Developer[] = [];
+
+  developers = developers.filter((developer) => {
+    if (callbackCondition(developer)) {
+      deleted.push(developer);
+      return false;
+    }
+
+    return true;
+  });
+
+  return deleted;
+}
+
+function sortDevelopersByEmployementAndAge(
+  isEmployed: boolean,
+  ageAscending = true,
+) {
+  developers.sort(function compare(firstDeveloper, secondDeveloper) {
+    if (
+      firstDeveloper.isEmployed === isEmployed &&
+      secondDeveloper.isEmployed === isEmployed
+    ) {
+      return ageAscending
+        ? secondDeveloper.age - firstDeveloper.age
+        : firstDeveloper.age - secondDeveloper.age;
+    }
+
+    return firstDeveloper.isEmployed === isEmployed ? -1 : 1;
+  });
+
+  return developers;
+}
+
+function listSkills(devId: number) {
+  const developer = developers.find(
+    ({ id: existingDevId }) => existingDevId === devId,
+  );
+
+  if (!developer) {
+    return;
+  }
+
+  return [...developer.skills];
+}
+
+function findDevelopersBySkill(searchWithskill: string) {
+  const devsWithSkill = developers.filter(({ skills }) => {
+    return skills.some((skill) => skill === searchWithskill);
+  });
+
+  return devsWithSkill;
+}
+
+function cloneDeveloper(devId: number) {
+  const developer = developers.find(
+    ({ id: existingDevId }) => existingDevId === devId,
```

Comment:
Why renaming again to existingDevId, directly use 

({ id }) => id === devId,

Reply:
My thought process was to make it a little more descriptive